docs/fundamentals/code-analysis/categories.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Code analysis rule categories
+description: Learn the different .NET code analysis rule categories.
+ms.date: 02/05/2021
+ms.topic: reference
+helpviewer_keywords:
+  - code analysis, categories
+---
+# Rule categories
+
+Each code analysis rule belongs to a category of rules. For example, design rules support adherence to the .NET design guidelines, and security rules help prevent security flaws. You can [disable or enable an entire category](configuration-options.md#scope) of rules. You can also [configure additional options](code-quality-rule-options.md#category-of-rules) on a per-category basis.
+
+The following table shows the different code analysis rule categories and provides a link to the rules in each category.
+
+| Category | Description |
+| - | - |
+| [Design rules](quality-rules/design-warnings.md) | Design rules support adherence to the [.NET Framework Design Guidelines](../../standard/design-guidelines/index.md). |
+| [Documentation rules](quality-rules/documentation-warnings.md) | Documentation rules support writing well-documented libraries through the correct use of XML documentation comments for externally visible APIs. |
+| [Globalization rules](quality-rules/globalization-warnings.md) | Globalization rules support world-ready libraries and applications. |
+| [Portability and interoperability rules](quality-rules/interoperability-warnings.md) | Portability rules support portability across different platforms. Interoperability rules support interaction with COM clients. |
+| [Maintainability rules](quality-rules/maintainability-warnings.md) | Maintainability rules support library and application maintenance. |
+| [Naming rules](quality-rules/naming-warnings.md) | Naming rules support adherence to the naming conventions of the .NET design guidelines. |
+| [Performance rules](quality-rules/performance-warnings.md) | Performance rules support high-performance libraries and applications. |
+| [Publish rules](quality-rules/publish-warnings.md) | Publish rules support single-file applications. |
+| [Reliability rules](quality-rules/reliability-warnings.md) | Reliability rules support library and application reliability, such as correct memory and thread usage. |
+| [Security rules](quality-rules/security-warnings.md) | Security rules support safer libraries and applications. These rules help prevent security flaws in your program. |
+| [Style rules](style-rules/index.md) | Style rules support consistent code style in your codebase. These rules start with the "IDE" prefix. |
```

Comment:
I included style rules as a category. @mavasani please confirm this is correct.

Reply:
Yes, this looks good.